warp/backenddb.go

```diff
@@ -0,0 +1,267 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/prefixdb"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/utils/linkedhashmap"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+)
+
+var _ WarpDb = &warpDb{}
+
+var (
+	tsPrefix			= []byte("t")
+	messageIDPrefix		= []byte("m")
+)
+
+var (
+	keyExistsError = errors.New("key exists")
+)
+
+const (
+	tsSize 			= 8
+	messageIDSize 	= 32
+
+	Nanosecond 		= uint64(time.Nanosecond)
+	Microsecond 	= uint64(time.Microsecond)
+	Millisecond 	= uint64(time.Millisecond)
+	Second 			= uint64(time.Second)
+	Minute			= uint64(time.Minute)
+	Hour			= uint64(time.Hour)
+	Day				= Hour * 24
+	Week			= Day * 7
+)
+
+type prunedMap linkedhashmap.LinkedHashmap[int, []byte]
+
+func NewPrunedMap() prunedMap {
+	return linkedhashmap.New[int, []byte]()
+}
+
```

Comment:
I think we can probably get rid of this too in favor of using `linkedhashmap` directly